src/cogent3/core/location.py

```diff
@@ -1099,7 +1090,1101 @@ def RangeFromString(string, delimiter=","):  # pragma: no cover
+        has shape (0, 2).
+        """
+        starts, ends = _gap_spans(self.gap_pos, self.cum_gap_lengths)
+        result = numpy.array([starts, ends]).T
+        if not len(result):
+            result = result.reshape((0, 2))
+        return result
+
+    def merge_maps(self, other, parent_length: Optional[int] = None) -> "IndelMap":
+        """merge gaps of other with self
+
+        Parameters
+        ----------
+        indel_map
+            instance for same sequence
+        parent_length
+            overrides property
+        """
+        unique_pos = numpy.union1d(self.gap_pos, other.gap_pos)
+        gap_lengths = numpy.zeros(unique_pos.shape, dtype=self.cum_gap_lengths.dtype)
+        self_lengths = self.get_gap_lengths()
+        other_lengths = other.get_gap_lengths()
+        _update_lengths(unique_pos, gap_lengths, self.gap_pos, self_lengths)
+        _update_lengths(unique_pos, gap_lengths, other.gap_pos, other_lengths)
+        parent_length = parent_length or self.parent_length
+        return self.__class__(
+            gap_pos=unique_pos,
+            gap_lengths=gap_lengths,
+            parent_length=parent_length,
+        )
+
+    def joined_segments(self, coords: SeqCoordTypes) -> "IndelMap":
+        """returns new map with disjoint gapped segments joined
+
+        Parameters
+        ----------
+        coords
+            sequence insert gap coordinates [(gap start, gap end), ...]
+        Returns
+        -------
+
+        """
+        coords = list(sorted(coords))
+        # using a dict here because joining can produce a gap merge
+        gaps = {}
+        cum_length = 0
+        cum_parent_length = 0
+        dtype = self.gap_pos.dtype
+        for start, end in coords:
+            im = self[start:end]
+            for i in range(im.num_gaps):
+                pos = im.gap_pos[i] + cum_parent_length
+                if pos in gaps:
+                    gaps[pos] += im.cum_gap_lengths[i]
+                else:
+                    gaps[pos] = im.cum_gap_lengths[i] + cum_length
+            cum_parent_length += im.parent_length
+            if im.num_gaps:
+                cum_length += im.cum_gap_lengths[-1]
+        gap_pos = numpy.empty(len(gaps), dtype=dtype)
+        cum_lengths = numpy.empty(len(gaps), dtype=dtype)
+        for i, (pos, length) in enumerate(sorted(gaps.items())):
+            gap_pos[i] = pos
+            cum_lengths[i] = length
+        return self.__class__(
+            gap_pos=gap_pos,
+            cum_gap_lengths=cum_lengths,
+            parent_length=cum_parent_length,
+        )
+
+    def nucleic_reversed(self) -> "IndelMap":
+        """map for a sequence that has itself been reversed and complemented
+
+        Notes
+        -----
+        discards reverse attribute on both spans and self
+        """
+        new_pos = self.gap_pos.copy()
+        lengths = self.get_gap_lengths()
+
+        if len(new_pos):
+            new_pos = self.parent_length - new_pos
```

Comment:
This feels off to me, `self.parent_length` refers to the number of non-gaps no?

Reply:
When we reverse a nucleic acid sequence, the gaps shift. For example with
```
#   01  2
seq AC--G
```
the gap is at sequence index 2. Now when we reverse (and I'm doing the reverse complement of the DNA sequence for "completeness"
```
#   0  12
seq C--GT
```
The gap sequence position is now 3-2=1

Reply:
What about a case like:
```
#   0 1 2
seq A-A-A
```
which would have gap positions at 1 and 3. We would expect the gap positions of the reverse complement
```
#   0 1 2
seq T-T-T
```
to still be 1 and 3, but the current algorithm would return 3-3=0 and 3-1=2 right?

I got the impression that this function should rely on the length of the entire sequence, as well as the length of the gaps (as we care about the position of the last character in the gap, not the first).

Reply:
As another example for
```
-A-AA----A
```
we should expect [1, 7, 9], but I believe we would instead get [-1, 2, 4]

Reply:
As another example, for

```
#    0 12    3
seq -A-AA----A
```

we should expect [1,7,9] but I believe we end up getting [-1,2,4]

Reply:
Disregard the above, I was indexing the gap positions incorrectly. For the first, the gap positions are originally [1,2] not [1,3]. For the second it is [0, 1, 3] for the original gap positions, not [0, 2, 5]